duckduckgo/Android
# delay app start for referrer data

Task/Issue URL: https://app.asana.com/0/72649045549333/1132602538216221/f
Tech Design URL: 
CC: 

**Description**:
Introduces a window for installation referrer data to arrive before showing the onboarding screen on first launch. Similarly, waits for referrer information to be available before eagerly initialising the ATB value. The max delay this should introduce is `1.5s`, and in the vast majority of cases it'll be significantly less than that (https://app.asana.com/0/72649045549333/1152651113794753/f)

**Steps to test this PR**:
Build and install locally to test whatever you want, but to test referrer data, see instructions in https://app.asana.com/0/72649045549333/1153635725536177/f

1. Perform a Fresh Install APK through Play Store (see link above), using referrer data 
1. Verify the onboarding screen is shown in an acceptably quick manner
1. Verify `pixel:m.rfa.*` fires (within a few seconds) and contains corresponding suffix (e.g., `XX` if using the link in the asana task)
1. Kill the app; restart it. Verify the app launches acceptably quickly
1. Use the 🔥 button; verify the app restarts acceptably quickly

1. Install the current prod app from the Play Store
1. Install the APK over the top to test updates work ok.

If you have any especially non-standard devices to hand, please test the APK runs ok on them. For example:

1. Amazon Fire tablets
1. Any other Android ROM which **doesn't have the Play Store**
1. Old / slow devices



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
